### PR TITLE
Support format json5

### PR DESCRIPTION
**Pull request by up9cloud**
😄

### CODE REVIEW

File: examples/glob/src/main.rs [hunk 1]
```diff
@@ -13,40 +13,58 @@ fn main() {
...
         // File::with_name(..) is shorthand for File::from(Path::new(..))
-        .merge(File::with_name("conf/00-default.toml")).unwrap()
-        .merge(File::from(Path::new("conf/05-some.yml"))).unwrap()
-        .merge(File::from(Path::new("conf/99-extra.json"))).unwrap();
+        .merge(File::with_name("conf/00-default.toml"))
+        .unwrap()
+        .merge(File::from(Path::new("conf/01.json")))
+        .unwrap()
+        .merge(File::from(Path::new("conf/99-extra.toml")))
+        .unwrap();
```
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:21]: Why did you change this?

File: examples/glob/src/main.rs [hunk 2]
```diff
@@ -13,40 +13,58 @@ fn main() {
...
 
     // Print out our settings (as a HashMap)
-    println!("\n{:?} \n\n-----------",
-             settings.try_into::<HashMap<String, String>>().unwrap());
+    println!(
+        "\n{:?} \n\n-----------",
+        settings
+            .try_into::<HashMap<String, config::Value>>()
+            .unwrap()
+    );
```
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:29]: Why did you change this?

File: examples/glob/src/main.rs [hunk 3]
```diff
@@ -13,40 +13,58 @@ fn main() {
...
     let mut settings = Config::default();
     settings
-        .merge(vec![File::with_name("conf/00-default.toml"),
-                    File::from(Path::new("conf/05-some.yml")),
-                    File::from(Path::new("conf/99-extra.json"))])
+        .merge(vec![
+            File::with_name("conf/00-default.toml"),
+            File::from(Path::new("conf/01.json")),
+            File::from(Path::new("conf/99-extra.toml")),
+        ])
```
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:40]: And this?

File: examples/glob/src/main.rs [hunk 4]
```diff
@@ -13,40 +13,58 @@ fn main() {
...
 
     // Print out our settings (as a HashMap)
-    println!("\n{:?} \n\n-----------",
-             settings.try_into::<HashMap<String, String>>().unwrap());
+    println!(
+        "\n{:?} \n\n-----------",
+        settings
+            .try_into::<HashMap<String, config::Value>>()
+            .unwrap()
+    );
```
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:49]: And this?

File: examples/glob/src/main.rs [hunk 5]
```diff
@@ -13,40 +13,58 @@ fn main() {
...
-        .merge(glob("conf/*")
-                   .unwrap()
-                   .map(|path| File::from(path.unwrap()))
-                   .collect::<Vec<_>>())
+        .merge(
+            glob("conf/*")
+                .unwrap()
+                .map(|path| File::from(path.unwrap()))
+                .collect::<Vec<_>>(),
+        )
```
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:61]: And this?

File: examples/glob/src/main.rs [hunk 6]
```diff
@@ -13,40 +13,58 @@ fn main() {
...
         .unwrap();
 
-    // Print out our settings (as a HashMap)
-    println!("\n{:?} \n\n-----------",
-             settings.try_into::<HashMap<String, String>>().unwrap());
+    // Print out our settings
+    println!(
+        "\n{:?} \n\n-----------\n{:?} \n\n-----------",
+        settings.cache.clone(), // detail with Value.origin
```
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:67]: Why is a `clone()` call necessary to print something?

File: src/file/format/hjson.rs [hunk 7]
```diff
@@ -2,15 +2,25 @@ use serde_hjson;
...
-    let value = from_hjson_value(uri, &serde_hjson::from_str(text)?);
+    let root = serde_hjson::from_str(text)?;
+    match root {
+        serde_hjson::Value::Object(_) => (),
+        _ => {
+            return Err(Box::new(io::Error::new(
+                io::ErrorKind::InvalidData,
+                format!("Root of resource({:?}) is not a Table", uri),
+            )))
+        }
```
Comment (matthiasbeyer) [inline src/file/format/hjson.rs:21]: You can remove this scope I guess.

File: src/file/format/json.rs [hunk 8]
```diff
@@ -2,15 +2,25 @@ use serde_json;
...
-    let value = from_json_value(uri, &serde_json::from_str(text)?);
+    let root = serde_json::from_str(text)?;
+    match root {
+        serde_json::Value::Object(_) => (),
+        _ => {
+            return Err(Box::new(io::Error::new(
+                io::ErrorKind::InvalidData,
+                format!("Root of resource({:?}) is not a Table", uri),
+            )))
+        }
```
Comment (matthiasbeyer) [inline src/file/format/json.rs:21]: And this scope.

File: src/file/format/json.rs [hunk 9]
```diff
@@ -22,13 +32,15 @@ fn from_json_value(uri: Option<&String>, value: &serde_json::Value) -> Value {
...
-        },
+        serde_json::Value::Number(ref value) => {
+            if let Some(value) = value.as_i64() {
+                Value::new(uri, ValueKind::Integer(value))
+            } else if let Some(value) = value.as_f64() {
+                Value::new(uri, ValueKind::Float(value))
+            } else {
+                unreachable!();
+            }
+        }
```
Comment (matthiasbeyer) [inline src/file/format/json.rs:43]: This also seems like a formatting change only...

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
There seem to be some more unnecessary scopes I did not comment.

Also, the diff includes code format changes which have nothing to do with this PR, which should be filed in another PR (or which are already filed in #155).

File: examples/glob/src/main.rs (hunk 1 above)
Comment (up9cloud) [reply examples/glob/src/main.rs:21]: Well, I just want to make it more complete, so I added files including all format we supported.
- 05-some.yml -> 01.json: I'm badly at naming, so I just named them all to [0-9].ext,  e.q. 04.ini...
- 99-extra.json -> 99-extra.toml: just for showing how json file works, so I chose 00-default to overwrite.

File: examples/glob/src/main.rs (hunk 2 above)
Comment (up9cloud) [reply examples/glob/src/main.rs:29]: Because of new files with nested structure, so I need change it to `config::Value`

File: examples/glob/src/main.rs (hunk 3 above)
Comment (up9cloud) [reply examples/glob/src/main.rs:40]: Some reason as above.

File: examples/glob/src/main.rs (hunk 4 above)
Comment (up9cloud) [reply examples/glob/src/main.rs:49]: Because new files are with nested values, so I need change it to config::Value

File: examples/glob/src/main.rs (hunk 5 above)
Comment (up9cloud) [reply examples/glob/src/main.rs:61]: It's same code. It changed because rustfmt.

File: examples/glob/src/main.rs (hunk 6 above)
Comment (up9cloud) [reply examples/glob/src/main.rs:67]: There are nested values in new files.

File: src/file/format/hjson.rs (hunk 7 above)
Comment (up9cloud) [reply src/file/format/hjson.rs:21]: I just followed the original TODO, if it's not necessary, please let me know.

File: src/file/format/json.rs (hunk 8 above)
Comment (up9cloud) [reply src/file/format/json.rs:21]: Same as above. I just followed the original TODO, if it's not necessary, please let me know.

File: src/file/format/json.rs (hunk 9 above)
Comment (up9cloud) [reply src/file/format/json.rs:43]: Yes, rustfmt did.

**Comment by matthiasbeyer** [comment]
Are you still interested in this? 

If yes, I created a maintenance fork (read [here](https://www.reddit.com/r/rust/comments/m43p3p/configmaint_maintenance_fork_of_config/)).
Feel free to submit your patches! :smile:

File: README.md [hunk 10]
```diff
@@ -8,18 +9,19 @@
...
- - Read from [JSON], [TOML], [YAML], [HJSON], [INI] files
- - Read from environment
- - Loosely typed — Configuration values may be read in any supported type, as long as there exists a reasonable conversion
- - Access nested fields using a formatted path — Uses a subset of JSONPath; currently supports the child ( `redis.port` ) and subscript operators ( `databases[0].name` )
+- Set defaults
+- Set explicit values (to programmatically override)
+- Read from [JSON], [TOML], [YAML], [HJSON], [INI], [JSON5] files
+- Read from environment
+- Loosely typed — Configuration values may be read in any supported type, as long as there exists a reasonable conversion
+- Access nested fields using a formatted path — Uses a subset of JSONPath; currently supports the child ( `redis.port` ) and subscript operators ( `databases[0].name` )
```
Comment (matthiasbeyer) [inline README.md:17]: This is an unnecessary format change, please do not include it in this PR.

File: README.md [hunk 11]
```diff
@@ -28,11 +30,12 @@
...
- - `json` - Adds support for reading JSON files
- - `hjson` - Adds support for reading HJSON files
- - `yaml` - Adds support for reading YAML files
- - `toml` - Adds support for reading TOML files
+- `ini` - Adds support for reading INI files
+- `json` - Adds support for reading JSON files
+- `hjson` - Adds support for reading HJSON files
+- `yaml` - Adds support for reading YAML files
+- `toml` - Adds support for reading TOML files
+- `json5` - Adds support for reading JSON5 files
```
Comment (matthiasbeyer) [inline README.md:38]: This is an unnecessary format change, please do not include it in this PR.

File: examples/glob/conf/00-default.toml [hunk 12]
```diff
@@ -1 +1,9 @@
-debug = false
+# comment
+seq = "00"
+
+[toml]
+s = "s"
+b = false
+i = 1
+f = 0.1
+a = ["t", "o", "m", "l"]
```
Comment (matthiasbeyer) [inline examples/glob/conf/00-default.toml:9]: I'm not sure why you're including changes to a toml file in a PR for json5.

File: examples/glob/conf/01.json [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+        "f": 0.1,
+        "a": [
+            null,
+            "s",
+            true,
+            1,
+            0.1
+        ]
+    }
+}
```
Comment (matthiasbeyer) [inline examples/glob/conf/01.json:17]: I'm not sure why you're including changes to a json file in a PR for json5.

File: examples/glob/conf/02.yaml [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+    s: s
+    b: false
+    i: 1
+    f: 0.1
+    a:
+        - null
+        - s
+        - true
+        - 1
+        - 0.1
```
Comment (matthiasbeyer) [inline examples/glob/conf/02.yaml:15]: I'm not sure why you're including changes to a yaml file in a PR for json5.

File: examples/glob/conf/02.yml [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+    s: s
+    b: false
+    i: 1
+    f: 0.1
+    a:
+        - null
+        - s
+        - true
+        - 1
+        - 0.1
```
Comment (matthiasbeyer) [inline examples/glob/conf/02.yml:15]: I'm not sure why you're including changes to a yaml file in a PR for json5.

File: examples/glob/conf/03.hjson [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+        f: 0.1
+        a: [
+            null
+            "s"
+            true
+            1
+            0.1
+        ]
+    }
+}
```
Comment (matthiasbeyer) [inline examples/glob/conf/03.hjson:20]: I'm not sure why you're including changes to a hjson file in a PR for json5.

File: examples/glob/conf/04.ini [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+n = null
+s = s
+b = false
+i = 1
+f = 0.1
+; a[] = null
+; a[] = s
+; a[] = true
+; a[] = 1
+; a[] = 0.1
```
Comment (matthiasbeyer) [inline examples/glob/conf/04.ini:15]: I'm not sure why you're including changes to a ini file in a PR for json5.

File: examples/glob/conf/05-some.yml [hunk 18]
```diff
@@ -1,2 +0,0 @@
-secret: THIS IS SECRET
-debug: true
```
Comment (matthiasbeyer) [inline examples/glob/conf/05-some.yml:2]: I'm not sure why you're including changes to a yaml file in a PR for json5.

File: examples/glob/conf/99-extra.json [hunk 19]
```diff
@@ -1,5 +0,0 @@
-{
-  "that": 3,
-  "this": 1230,
-  "key": "sdgnjklsdjklgds"
-}
```
Comment (matthiasbeyer) [inline examples/glob/conf/99-extra.json:5]: I'm not sure why you're including changes to a json file in a PR for json5.

File: examples/glob/conf/99-extra.toml [hunk 20]
```diff
@@ -0,0 +1,2 @@
+seq = "99"
+extra = true
```
Comment (matthiasbeyer) [inline examples/glob/conf/99-extra.toml:2]: I'm not sure why you're including changes to a toml file in a PR for json5.

File: examples/glob/src/main.rs (hunk 1 above)
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:21]: This is an unnecessary formatting change, please do not include it in this PR.

File: examples/glob/src/main.rs (hunk 2 above)
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:29]: This is partly a formatting change, please include only the relevant bits in this PR.

File: examples/glob/src/main.rs (hunk 3 above)
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:40]: This is partly a formatting change, please include only the relevant bits in this PR.

File: examples/glob/src/main.rs (hunk 4 above)
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:49]: This is partly a formatting change, please include only the relevant bits in this PR.

File: examples/glob/src/main.rs (hunk 5 above)
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:61]: This is partly an unnecessary formatting change, please do not include it in this PR.

File: examples/glob/src/main.rs (hunk 6 above)
Comment (matthiasbeyer) [inline examples/glob/src/main.rs:67]: This is partly a formatting change, please include only the relevant bits in this PR.

File: src/file/format/hjson.rs [hunk 21]
```diff
@@ -1,17 +1,41 @@
...
+        serde_hjson::Value::Bool(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Bool(value)))
+        }
+        serde_hjson::Value::Object(_) => (),
+        serde_hjson::Value::Array(_) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Seq))
+        }
+        serde_hjson::Value::Null => return Err(ConfigError::invalid_root(uri, Unexpected::Unit)),
+    };
```
Comment (matthiasbeyer) [inline src/file/format/hjson.rs:37]: I guess we can simplify this whole `match` to something like this:

```
let unexp_err = match {
    serde_hjson::Value::Object(_) => None,
    serde_hjson::Value::String(v) => Unexpected::Str(value),
    /* and the other Unexpected instances */
};

if let Some(unex) = unexp_err {
    return Err(ConfigError::unvalid_root(uri, unex))
}
```

which would reduce the mental load on this one a lot, IMO. :smile:

File: src/file/format/json.rs [hunk 22]
```diff
@@ -1,17 +1,39 @@
...
+                Unexpected::Integer(value.as_i64().unwrap_or(0)),
+            ));
+        }
+        serde_json::Value::Bool(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Bool(value)))
+        }
+        serde_json::Value::Object(_) => (),
+        serde_json::Value::Array(_) => return Err(ConfigError::invalid_root(uri, Unexpected::Seq)),
+        serde_json::Value::Null => return Err(ConfigError::invalid_root(uri, Unexpected::Unit)),
+    };
```
Comment (matthiasbeyer) [inline src/file/format/json.rs:35]: Something like my previous comment can be done here, I guess!

File: src/file/format/json5.rs [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+        Val::String(value) => return Err(ConfigError::invalid_root(uri, Unexpected::Str(value))),
+        Val::Integer(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Integer(value)))
+        }
+        Val::Float(value) => return Err(ConfigError::invalid_root(uri, Unexpected::Float(value))),
+        Val::Boolean(value) => return Err(ConfigError::invalid_root(uri, Unexpected::Bool(value))),
+        Val::Object(_) => (),
+        Val::Array(_) => return Err(ConfigError::invalid_root(uri, Unexpected::Seq)),
+        Val::Null => return Err(ConfigError::invalid_root(uri, Unexpected::Unit)),
+    };
```
Comment (matthiasbeyer) [inline src/file/format/json5.rs:36]: ... and here.

File: src/file/format/json5.rs [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+        ValueKind::Table(map) => Ok(map),
+
+        _ => Ok(HashMap::new()),
+    }
+}
+
+fn from_json5_value(uri: Option<&String>, value: &Val) -> Value {
+    match *value {
+        Val::String(ref v) => Value::new(uri, ValueKind::String(v.clone())),
+
```
Comment (matthiasbeyer) [inline src/file/format/json5.rs:48]: please remove the blank lines between the match arms here.

File: src/file/format/json5.rs [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+        Val::Object(ref table) => {
+            let mut m = HashMap::new();
+
+            for (key, value) in table {
+                m.insert(key.clone(), from_json5_value(uri, value));
+            }
+
+            Value::new(uri, ValueKind::Table(m))
+        }
```
Comment (matthiasbeyer) [inline src/file/format/json5.rs:63]: I guess this scope would be nicer like this:

```rust
let m = table.into_iter()
    .map(|(k, v)| (k, from_json5_value(uri, v)))
    .collect::<HashMap<_, _>>();
Value::new(uri, ValueKind::Table(m))
```

But you'd have to remove the `ref` from `ref table`. And it would result in one less clone.

File: src/file/format/json5.rs [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+
+        Val::Array(ref array) => {
+            let mut l = Vec::new();
+
+            for value in array {
+                l.push(from_json5_value(uri, value));
+            }
+
+            Value::new(uri, ValueKind::Array(l))
+        }
```
Comment (matthiasbeyer) [inline src/file/format/json5.rs:73]: Same as the last comment applies here.

File: src/file/format/mod.rs [hunk 27]
```diff
@@ -39,9 +42,14 @@ pub enum FileFormat {
     /// HJSON (parsed with serde_hjson)
     #[cfg(feature = "hjson")]
     Hjson,
+
```
Comment (matthiasbeyer) [inline src/file/format/mod.rs:45]: This has nothing to do with the PR, please remove this change.

File: src/file/format/toml.rs [hunk 28]
```diff
@@ -10,7 +10,6 @@ pub fn parse(
     text: &str,
 ) -> Result<HashMap<String, Value>, Box<dyn Error + Send + Sync>> {
     // Parse a TOML value from the provided text
-    // TODO: Have a proper error fire if the root of a file is ever not a Table
```
Comment (matthiasbeyer) [inline src/file/format/toml.rs:13]: This has nothing to do with the PR, please remove this change.

File: src/file/format/yaml.rs [hunk 29]
```diff
@@ -5,6 +5,7 @@ use std::mem;
 
 use yaml_rust as yaml;
 
+use crate::error::{ConfigError, Unexpected};
```
Comment (matthiasbeyer) [inline src/file/format/yaml.rs:8]: I'm not sure why you're changing yaml stuff in a PR for json5, please remove this change.

File: src/file/format/yaml.rs [hunk 30]
```diff
@@ -20,8 +21,26 @@ pub fn parse(
...
+        yaml::Yaml::Integer(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Integer(value)))
+        }
+        yaml::Yaml::Boolean(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Bool(value)))
+        }
+        yaml::Yaml::Hash(_) => (),
+        yaml::Yaml::Array(_) => return Err(ConfigError::invalid_root(uri, Unexpected::Seq)),
+        _ => return Err(ConfigError::invalid_root(uri, Unexpected::Unit)),
+    };
```
Comment (matthiasbeyer) [inline src/file/format/yaml.rs:43]: I'm not sure why you're changing yaml stuff in a PR for json5, please remove this change.

File: src/ser.rs [hunk 31]
```diff
@@ -699,7 +699,7 @@ mod test {
 
     #[test]
     fn test_struct() {
-        #[derive(Debug, Serialize, Deserialize, PartialEq)]
+        #[derive(Debug, serde_derive::Deserialize, serde_derive::Serialize, PartialEq)]
```
Comment (matthiasbeyer) [inline src/ser.rs:702]: I don't see why this change was necessary?

File: tests/errors.rs [hunk 32]
```diff
@@ -28,7 +39,7 @@ fn test_error_parse() {
     assert_eq!(
         res.unwrap_err().to_string(),
         format!(
-            "invalid TOML value, did you mean to use a quoted string? at line 2 column 9 in {}",
+            "failed to parse datetime for key `error` at line 2 column 9 in {}",
```
Comment (matthiasbeyer) [inline tests/errors.rs:42]: How does this belong to this PR?

File: tests/errors.rs [hunk 33]
```diff
@@ -106,7 +117,7 @@ fn test_error_enum_de() {
 }
 
 #[test]
-fn error_with_path() {
+fn test_error_with_path() {
```
Comment (matthiasbeyer) [inline tests/errors.rs:120]: How does this belong to this PR?

File: tests/file_toml.rs [hunk 34]
```diff
@@ -88,7 +88,7 @@ fn test_error_parse() {
     assert_eq!(
         res.unwrap_err().to_string(),
         format!(
-            "invalid TOML value, did you mean to use a quoted string? at line 2 column 9 in {}",
+            "failed to parse datetime for key `error` at line 2 column 9 in {}",
```
Comment (matthiasbeyer) [inline tests/file_toml.rs:91]: How does this belong to this PR?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
So, I did a review of all changes in this PR.

Overall I have to say that this goes into the right direction, I guess. Still, I have quite a few remarks.

In general: Could you please rebase to master? This way the merge commits would vanish, which is particularly important to me, as it cleans the history quite a bit. This PR is based on a really old commit and I guess some of my remarks here only exist because of the merge, which makes this diff here appear to have changes which are made on master and not in this PR. I hope. If they are indeed from this PR, please make sure to file changes that do not conceptionally belong to the json5 feature in seperate PRs. But I'd bet, as said, that most of my remarks render invalid after your rebase! :+1: 

Also, please make clippy happy! :laughing:

File: src/file/format/json5.rs (hunk 25 above)
Comment (up9cloud) [reply src/file/format/json5.rs:64]: Well, I think iter or loop, they are just different coding style. (I was following the style from other format files, e.q. https://github.com/mehcode/config-rs/blob/master/src/file/format/hjson.rs#L34). :D I guess we could change coding style on next commit?
And I don't get your "one less clone" point... in my understanding, they are same? (I'm not very sure, but...)

**Comment by up9cloud** [comment]
Remove all nonrelative files, enjoy 😄

**Comment by matthiasbeyer** [comment]
Nice. Now please make it run successfully through CI, then I'll do another review.

File: tests/errors.rs (hunk 32 above)
Comment (up9cloud) [reply tests/errors.rs]: @matthiasbeyer It was weird when I ran test on my local environment, it was failed and I got the different error message, so I made this change to pass test. It seems CI has diff result. I reverted this change.

File: tests/file_toml.rs (hunk 34 above)
Comment (up9cloud) [reply tests/file_toml.rs]: Same as above, and I reverted

**Comment by up9cloud** [comment]
> Nice. Now please make it run successfully through CI, then I'll do another review.

All green now, :)

File: src/file/format/json5.rs (hunk 25 above)
Comment (up9cloud) [reply src/file/format/json5.rs:64]: I see, you want me move them from root. I was confused by your into_iter() because the fn was 
passing root by ref, sorry about that. I removed all clone() parts. enjoy :)

File: src/file/format/json5.rs (hunk 26 above)
Comment (up9cloud) [reply src/file/format/json5.rs:74]: solved, see above.

File: src/error.rs [hunk 35]
```diff
@@ -95,6 +95,18 @@ impl ConfigError {
         }
     }
 
+    // Have a proper error fire if the root of a file is ever not a Table
+    // TODO: for now only json5 checked, need to finish others
```
[the remark this answers is not in the document]
Comment (up9cloud) [reply src/error.rs:98]: Well, this TODO is from something like this: https://github.com/mehcode/config-rs/blob/266f504d9f23e192c03ef486f58a678847249b60/src/file/format/hjson.rs#L11

**Review by up9cloud — COMMENTED** [review]
clone() removed. enjoy :)

**Comment by matthiasbeyer** [comment]
Nice. But no need to merge master, please remove that, it makes review really hard.

**Comment by up9cloud** [comment]
@matthiasbeyer I guess what you meant is rebase, not remove.

I have to choose merge or rebase, otherwise it won't pass CI testing. 

I pushed the rebase version. enjoy.

**Comment by matthiasbeyer** [comment]
Thanks for your contribution! json5 support was merged in #206
